### PR TITLE
[quant][fx] Follow up fixes for qconfig validations for fixedqparams ops

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #81010

Summary:
This adds a few things on top of https://github.com/pytorch/pytorch/pull/80184,
1). node.target was assumed to be "tanh", torch.nn.Tanh etc. this PR handles that properly
2). adds FixedQParamsFakeQuantize support
3). extends the comparison function _partial_wrapper_equals to work with FakeQuantize.with_args(observer=...)

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D37735193](https://our.internmc.facebook.com/intern/diff/D37735193)

### CODE REVIEW

File: torch/ao/quantization/fx/convert.py [hunk 1]
```diff
@@ -473,7 +474,7 @@ def convert_custom_module(
         node: Node,
         graph: Graph,
         modules: Dict[str, torch.nn.Module],
-        custom_module_class_mapping: Dict[Callable, Callable],
+        custom_module_class_mapping: Dict[QuantType, Dict[Type, Type]],
```
Comment (andrewor14) [inline torch/ao/quantization/fx/convert.py:477]: Is this a related change?

**Review by andrewor14 — APPROVED** [review]
Looks good. Thanks for following up on this!

File: torch/ao/quantization/fx/convert.py (hunk 1 above)
Comment (jerryzh168) [reply torch/ao/quantization/fx/convert.py:477]: oh this is to fix the type checking

File: torch/ao/quantization/fx/prepare.py [hunk 2]
```diff
@@ -1334,17 +1336,33 @@ def _validate_fixed_qparams_qconfigs(model: GraphModule, qconfig_map: Dict[str,
...
+            if qconfig is None:
+                bad_observer = False
+            else:
+                for observer_ctr in allowed_observer_ctrs + [_FIXED_QPARAMS_OP_TO_OBSERVER[module_type_or_function_or_method]]:
+                    if _partial_wrapper_equals(
+                            qconfig.activation,
+                            FixedQParamsFakeQuantize.with_args(observer=observer_ctr)):
+                        bad_observer = False
+                    if _partial_wrapper_equals(qconfig.activation, observer_ctr):
+                        bad_observer = False
```
[the remark this answers is not in the document]
Comment (jerryzh168) [reply torch/ao/quantization/fx/prepare.py:1360]: sure

File: torch/ao/quantization/qconfig.py [hunk 3]
```diff
@@ -416,12 +418,28 @@ def configure_constructor_to_put_obs_on_module_device(original_constructor):
...
 def _partial_wrapper_equals(obs1: _PartialWrapper, obs2: _PartialWrapper):
     """
     Return whether the two partial wrappers are equal,
     """
     # functools.partial has no __eq__ operator defined so '==' defaults to 'is'
-    return obs1.p.func == obs2.p.func and obs1.p.args == obs2.p.args and obs1.p.keywords == obs2.p.keywords
+    obs1_keywords = copy.copy(obs1.p.keywords)
+    obs2_keywords = copy.copy(obs2.p.keywords)
+    keywords_equal = True
+    # compare observer with _partial_wrapper_equals as well since direct compare would fail
```
[the remark this answers is not in the document]
Comment (jerryzh168) [reply torch/ao/quantization/qconfig.py:436]: right, forgot to update the comment

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37735193).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37735193).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37735193).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37735193).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37735193).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37735193).
